### ISSUE TITLE
published crd schema helmrelease-helm-v2beta1.json incorrectly shows spec/chart as required

### ISSUE THREAD

**Issue by nralbers** [fluxcd/flux2#5050]
### Describe the bug

The published standalone schema for helmrelease-helm-v2beta1.json will flag a release using a chartRef instead of a chart as invalid.

### Steps to reproduce

1. Create Helmrelease using OCI repository as a chartref 
2. Run https://github.com/fluxcd/flux2-kustomize-helm-example/blob/main/scripts/validate.sh on cluster
3. validation will fail on helmrelease stating that chart is a required property of spec

Example helmrelease
```
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: external-dns
  namespace: external-dns
spec:
  chartRef:
    kind: OCIRepository
    name: external-dns
    namespace: flux-system
  interval: 15m
  timeout: 5m
  releaseName: external-dns
  values: 
    ...
```

### Expected behavior

validation json should allow either chart or chartref

So
```json
"spec": {
  ...
   "required": [
            "chart",
            "interval"
          ],
         
} 
```

becomes
```json
"spec": {
  ...
   "required": [
            "interval"
          ],
          "oneOf": [
              {
                  "required": [
                      "chart"
                  ]
              },
              {
                  "required":[
                      "chartRef"
                  ]
              }
          ],
} 
```

### OS / Distro

N/A

### Flux version

v2.4.0

### Flux check

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by stefanprodan** [comment]
The `chartRef` is only supposed in v2

**Comment by gecube** [comment]
Did some archaeology on this one, since the issue is still open.

**TL;DR:** the published `helmrelease-helm-v2beta1.json` was not wrong — it was a faithful 1:1 mirror of the actual `v2beta1` CRD admission schema. As of Flux v2.4.0 the API server itself rejects a `v2beta1` HelmRelease that has only `chartRef`. Relaxing the standalone schema to `oneOf` would have made offline validation pass for manifests the cluster then rejects at apply time. And in current Flux both deprecated versions are gone entirely, so there is nothing left to fix — this can probably be closed.

Live check against the Flux v2.4.0 CRD (helm-controller v1.1.0), server-side dry-run of the exact chartRef-only spec from the issue at all three served versions:

```
=== v2beta1 ===
The HelmRelease "external-dns-v2beta1" is invalid: spec.chart: Required value
=== v2beta2 ===
helmrelease.helm.toolkit.fluxcd.io/external-dns-v2beta2 created (server dry run)
=== v2 ===
helmrelease.helm.toolkit.fluxcd.io/external-dns-v2 created (server dry run)
```

So kubeconform flagging `spec.chart` as required for `v2beta1` was accurately predicting what the API server would do.

**Why chartRef-only `v2beta1` manifests may have *appeared* to work before v2.4.0** (probably how such manifests came to exist in the first place):

- The v2beta1 API intent was always "chart required": [`Chart` carries a `+required` marker, and `ChartRef` is documented as *"provisional to the v2 API, and not actively used by v2beta1 HelmReleases"*](https://github.com/fluxcd/helm-controller/blob/v1.1.0/api/v2beta1/helmrelease_types.go#L70-L81).
- But helm-controller v1.0.x (Flux v2.3.x) generated its CRDs with controller-gen v0.15.0, which did not yet understand the plain k8s-style `+required` marker on a field with `omitempty` — so the shipped v2beta1 schema accidentally had `required: [interval]` only, and a chartRef-only `v2beta1` manifest was admitted. Since the CRD uses the default conversion strategy (`None`) with storage version `v2`, the controller then reconciled it with full v2 semantics, i.e. it genuinely worked.
- helm-controller v1.1.0 (Flux v2.4.0) bumped controller-gen to v0.16.1, which added support for the `+required` marker (kubernetes-sigs/controller-tools#944) — `spec.chart` became required at v2beta1 again, matching the documented API intent, and the published JSON schema followed.

One small correction to the comment above: `chartRef` was in fact usable via `v2beta2` as well, not only `v2` — v2beta2's [`Chart` is explicitly `+optional` with the same "either chart or chartRef must be set" CEL rule as v2](https://github.com/fluxcd/helm-controller/blob/v1.1.0/api/v2beta2/helmrelease_types.go#L79-L84). So the migration path at the time was: switch the manifest to `v2` (or `v2beta2`), not patch the JSON schema.

**Current state:**

- `v2beta1` was removed from the CRDs in helm-controller v1.4.0 (Flux v2.7.0) and `v2beta2` in v1.5.0 (Flux v2.8.0). The `crd-schemas.tar.gz` published with v2.9.2 ships only `helmrelease-helm-v2.json`, whose spec-level `required` is `["interval"]` — a chartRef-only HelmRelease validates fine.
- The `validate.sh` from flux2-kustomize-helm-example referenced in the repro steps has since been rewritten to use the `flux-schema` plugin with the ecosystem schema catalog instead of kubeconform + the release schema tarball, so the reported failure mode is no longer reachable via that script either.

cc @matheuscscp — I believe this one can be closed as working-as-intended at the time / obsolete now.
